src/webgpu/api/validation/encoding/cmds/clearBuffer.spec.ts

```diff
@@ -0,0 +1,196 @@
+export const description = `
+API validation tests for clearBuffer.
+`;
+
+import { makeTestGroup } from '../../../../../common/framework/test_group.js';
+import { kBufferUsages } from '../../../../capability_info.js';
+import { kMaxSafeMultipleOf8 } from '../../../../util/math.js';
+import { ValidationTest } from '../../validation_test.js';
+
+class F extends ValidationTest {
+  TestClearBuffer(options: {
+    buffer: GPUBuffer;
+    offset: number | undefined;
+    size: number | undefined;
+    isSuccess: boolean;
+  }): void {
+    const { buffer, offset, size, isSuccess } = options;
+
+    const commandEncoder = this.device.createCommandEncoder();
+    commandEncoder.clearBuffer(buffer, offset, size);
+
+    this.expectValidationError(() => {
+      commandEncoder.finish();
+    }, !isSuccess);
+  }
+}
+
+export const g = makeTestGroup(F);
+
+g.test('invalid_buffer')
+  .desc(`Test that clearing an error buffer fails.`)
+  .fn(async t => {
+    const errorBuffer = t.getErrorBuffer();
+
+    t.TestClearBuffer({
+      buffer: errorBuffer,
+      offset: 0,
+      size: 8,
+      isSuccess: false,
+    });
+  });
+
+g.test('default_args')
+  .desc(`Test that calling clearBuffer with a default offset and size is valid.`)
```

Comment:
Should also test with offset=undefined,size=value; and offset=value,size=undefined